doc/languages-frameworks/gradle.section.md

```diff
@@ -0,0 +1,203 @@
+# Gradle {#gradle}
+
+Gradle is a popular build tool for Java/Kotlin. Gradle itself doesn't
+currently provide tools to make dependency resolution reproducible, so
+nixpkgs has a proxy designed for intercepting Gradle web requests to
+record dependencies so they can be restored in a reproducible fashion.
+
+## Building a Gradle package {#building-a-gradle-package}
+
+A simple case is when the package doesn't have any dependencies. In that
+case, simply adding `gradle` to `nativeBuildInputs` may be enough:
+
+```nix
+stdenv.mkDerivation {
+  pname = "jextract";
+  version = "unstable-2023-11-27";
+
+  src = fetchFromGitHub {
+    owner = "openjdk";
+    repo = "jextract";
+    rev = "8730fcf05c229d035b0db52ee6bd82622e9d03e9";
+    hash = "sha256-Wct/yx5C0EjDtDyXNYDH5LRmrfq7islXbPVIGBR6x5Y=";
+  };
+
+  nativeBuildInputs = [
+    gradle
+    makeWrapper
+  ];
+
+  gradleFlags = [
+    "-Pllvm_home=${llvmPackages.libclang.lib}"
+    "-Pjdk21_home=${jdk21}"
+  ];
+
+  doCheck = true;
+
+  gradleCheckTask = "verify";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/path/to/jextract
+    cp -r ./build/jextract/* $out/path/to/jextract/
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    makeWrapper "$out/path/to/jextract/bin/jextract" "$out/bin/jextract"
+  '';
+}
+```
```

Comment:
This is the first and only complete-looking example anyone attempting to use gradle will see and use as a reference.

It does a bunch of non-standard stuff on the one hand (llvm home?) and yet doesn't do many things I'd consider common (mitmCache, passthru.updateDeps, gradleBuildTask, makeWrapper to wrap a jar).

I think it'd be better to have a rather complete example of all the things you may need to do first and foremost and leave the uncommon event of a package with no dependencies for an advanced example (or just a comment really).

You might be surprised but after not having worked with this PR for a week or so and attempting to use the docs as a guide, this has proven to be a rather significant roadblock.

Reply:
`makeWrapper` to wrap a jar is reasonable to add here, yes.

I don't think making the example have dependencies is that good of an idea because the next two paragraphs are important for understanding how to create and update deps.json. I think I could add additional emphasis on the fact the package doesn't have any dependencies.

Reply:
now that I simplified it by merging `fetchDeps` and `updateDeps`, this seems reasonable, I've done just that